### PR TITLE
[Fix] Add manual Timeout to Prevent SocketTimeoutException on Android

### LINKED ISSUE #295: SocketTimeoutException on Android
I got java.net.SocketTimeoutException during downloading the file an. on Android:

```
java.net.SocketTimeoutException: timeout
2020-05-15 10:24:23.852 10660-10973/com.myapp W/System.err:     at com.android.okhttp.okio.Okio$3.newTimeoutException(Okio.java:212)
2020-05-15 10:24:23.853 10660-10973/com.myapp  W/System.err:     at com.android.okhttp.okio.AsyncTimeout.exit(AsyncTimeout.java:250)
2020-05-15 10:24:23.853 10660-10973/com.myapp W/System.err:     at 
```

I see that the reason is in your **DownloadWorker.java** class. Timeout is 15 seconds. How to increase it?

_httpConn.setConnectTimeout(15000);
httpConn.setReadTimeout(15000);_

https://github.com/fluttercommunity/flutter_downloader/blob/master/android/src/main/java/vn/hunghd/flutterdownloader/DownloadWorker.java

Thank you.

### PR DESCRIPTION
**Pull request by SebghatYusuf**
This will fix #295

### CODE REVIEW

**Comment by bartekpacia** [comment]
Thanks for contributing.

I'd prefer to have this timeout configurable, can you do this? Preferably both for Android and iOS.

**Comment by SebghatYusuf** [comment]
> Thanks for contributing.
> 
> I'd prefer to have this timeout configurable, can you do this? Preferably both for Android and iOS, but if you don't want to do iOS, that's fine as well :)

Ok sure, I'll look into this.

**Comment by SebghatYusuf** [comment]
@bartekpacia 
The PR is updated and now it's configurable. please review. 
Thanks :)

File: lib/src/downloader.dart [hunk 1]
```diff
@@ -90,6 +90,8 @@ class FlutterDownloader {
   /// external storage. If you want to save the file in the public Downloads
   /// directory instead, set [saveInPublicStorage] to true. In that case,
   /// [savedDir] will be ignored.
+  ///
+  /// [timeout] is added for android devices to manually change connection timeout.
```
Comment (bartekpacia) [inline lib/src/downloader.dart:94]: This section is already titled `Android-only` (see line 76). 

```suggestion
  /// [timeout] is the HTTP connection timeout.
```

File: lib/src/downloader.dart [hunk 2]
```diff
@@ -282,6 +288,7 @@ class FlutterDownloader {
   /// **parameters:**
   ///
   /// * `taskId`: unique identifier of a failed download task
+  /// * `timeout`: http request connection timeout.
```
Comment (bartekpacia) [inline lib/src/downloader.dart:291]:
```suggestion
  /// * `timeout`: http request connection timeout. Android only.
```

File: lib/src/downloader.dart [hunk 3]
```diff
@@ -290,13 +297,15 @@ class FlutterDownloader {
   static Future<String?> retry({
     required String taskId,
     bool requiresStorageNotLow = true,
+    int timeout = 15000, // Works on Android only
```
Comment (bartekpacia) [inline lib/src/downloader.dart:300]:
```suggestion
    int timeout = 15000,
```

**Review by bartekpacia — COMMENTED** [review]
Looks good. Just a few micro changes. Thanks for contributing!

I'll approve once requested changes are done.

**Comment by SebghatYusuf** [comment]
@bartekpacia  
Changes applied.

**Comment by bartekpacia** [comment]
Thanks!

**Comment by IrisBC** [comment]
Thanks for this feature!
I am trying to provide a timeout in version 1.9.1 (tried also 1.9.0) which as I understand already have the merged changes.
However flutter does not seem to recognize the new parameter:
```
  static Future<String?> enqueue({
    required String url,
    required String savedDir,
    String? fileName,
    Map<String, String> headers = const {},
    bool showNotification = true,
    bool openFileFromNotification = true,
    bool requiresStorageNotLow = true,
    bool saveInPublicStorage = false,
  }) async {
...
```
Which is the status of this development? In case I should be able to use it, how should I do so?

Cheers,

**Comment by bartekpacia** [comment]
Hi, it looks like it's a small bug – the PR author could've forgotten to add this parameter :D 

I'll be grateful for a fix :)

**Comment by IrisBC** [comment]
Hi!
I am not sure, here: https://github.com/SebghatYusuf/flutter_downloader/blob/master/lib/src/downloader.dart the parameter is included as optional, but then using it in Flutter whitin andriod studio is not recognizing it in my case. 
Am I looking correctly?

Maybe some of my config is incorrect.

**Comment by bartekpacia** [comment]
Well that's strange. Can you show your pubspec.yaml?

**Comment by IrisBC** [comment]
I had version incampoatibilities with Kotlin versions. Will look at it more detailed. Thanks for the quick answer, hopefully i didnt take you too much time.

**Comment by bartekpacia** [comment]
No problem :) happy to help

**Pull request merged**